platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java

```diff
@@ -319,7 +319,13 @@ private void newSignedStateBeingTracked(final SignedState signedState, final Sou
         if (source == SourceOfSignedState.DISK) {
             signedStateFileManager.registerSignedStateFromDisk(signedState);
         } else {
-            signedStateFileManager.determineIfStateShouldBeSaved(signedState);
+            signedStateFileManager.determineIfStateShouldBeSaved(signedState, source);
+        }
+        if (source == SourceOfSignedState.RECONNECT && stateConfig.saveReconnectStateToDisk()) {
+            // a state received from reconnect should be saved to disk, but the method stateHasEnoughSignatures will not
+            // be called for it by the signed state manager, so we need to call it here
+            // we only call this method if the behaviour is enabled to retain the same behaviour as before
+            stateHasEnoughSignatures(signedState);
```

Comment:
Perhaps we should just directly call `signedStateFileManager.saveSignedStateToDisk(signedState);`. If we do that, then we won't have to change any of the logic in `shouldSaveToDisk()`.

Reply:
yeah, but that would mean scattering responsibility even more. This way, at least, there is a single method that determines if a state should be saved to disk

Reply:
Makes sense. I'm not 100% convinced, but this is not a big enough deal for me to block this PR moving forward. You can resolve this request.